com.unity.multiplayer.mlapi/Tests/Runtime/NetworkManagerHelper.cs

```diff
@@ -0,0 +1,204 @@
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+using NUnit.Framework;
+using MLAPI.SceneManagement;
+using MLAPI.Transports.UNET;
+
+namespace MLAPI.RuntimeTests
+{
+    /// <summary>
+    /// Helper class to instantiate a NetworkManager
+    /// This also provides the ability to:
+    /// --- instantiate GameObjects with NetworkObject components that returns a Guid for accessing it later.
+    /// --- add NetworkBehaviour components to the instantiated GameObjects
+    /// --- spawn a NetworkObject using its parent GameObject's Guid
+    /// Call StartNetworkManager in the constructor of your runtime unit test class.
+    /// Call ShutdownNetworkManager in the destructor  of your runtime unit test class.
+    ///
+    /// Includes a useful "BuffersMatch" method that allows you to compare two buffers (returns true if they match false if not)
+    /// </summary>
+    public static class NetworkManagerHelper
+    {
+        public static Transports.Tasks.SocketTasks s_StartHostSocketTasks { get; internal set; }
+        public static GameObject s_NetworkManagerObject { get; internal set; }
+
+        internal static Dictionary<Guid,GameObject> s_InstantiatedGameObjects = new Dictionary<Guid, GameObject>();
+
+        internal static Dictionary<Guid,NetworkObject> s_InstantiatedNetworkObjects = new Dictionary<Guid, NetworkObject>();
+
+        /// <summary>
+        /// Called upon the RpcQueueTests being instantiated.
+        /// This creates a NetworkManger,
+        /// </summary>
+        /// <returns></returns>
+        public static bool StartNetworkManager()
```

Comment:
Nit but this should probably be clear that it's starting a network manager **as host**

Reply:
+1

Reply:
I added the capability for one to start the NetworkManager in Host, Server, or client mode.  You should be able to switch modes during testing if one needed to test specific aspects of the Server or Client modes.

Reply:
Please review adjustment.